detection_rules/integrations.py

```diff
@@ -89,6 +162,49 @@ def find_least_compatible_version(package: str, integration: str,
     raise ValueError(f"no compatible version for integration {package}:{integration}")
 
 
+def find_compatible_version_window(package: str, integration: str,
+                                   rule_stack_version: str, packages_manifest: dict) -> Generator[str, None, None]:
+    """Finds least compatible version for specified integration based on stack version supplied."""
+
+    if not package:
+        raise ValueError("Package must be specified")
+
+    package_manifest = packages_manifest.get(package)
+    if package_manifest is None:
+        raise ValueError(f"Package {package} not found in manifest.")
+
+    integration_manifests = sorted(package_manifest.items(), key=lambda x: Version(str(x[0])))
+
+    # iterates through ascending integration manifests
+    # returns latest major version that is least compatible
+    for version, manifest in integration_manifests:
```

Comment:
some of this code looks redundant ..?

Reply:
Intentionally. We can break things out once we answer the big questions.

Reply:
With the latest refactor, this should have all changed either way.